lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -301,8 +302,9 @@ def error(response)
         response['result'].first['error']
       end
 
-      def current_date_time
-        DateTime.now.strftime('%Y-%m-%dT%H:%M:%S.%N+%H:%M')
+      def current_date_time(options = {})
+        time_zone = options[:merchant_time_zone] || 'Pacific Time (US & Canada)'
+        Time.now.in_time_zone(time_zone).strftime('%Y-%m-%dT%H:%M:%S.%3N+%H:%M')
```

Comment:
Good work on the `strftime` format, but is the `options[:merchant_time_zone]` something that the client has requested?

I think we should avoid exposing this option unless it is something that is specifically being asked for. The Rails method `in_time_zone` is just one of many ways to specify a time zone, and if we were going to implement this I think we would want to provide a list of acceptable values to pass, because if merchants start passing values like `'PDT'` or `'-06:00'` then we will get an error here.

I think it would be best to just let the customer pass their own timestamp in `options[:date_time]` and if they don't do that, they get whatever DateTime.now we provide.

Reply:
I think it is being asked to provide time in the merchant zone (not from where request is being made) and we have some documented time zones that could be used:
https://api.rubyonrails.org/classes/ActiveSupport/TimeZone.html

Reply:
Okay. Let's make sure we document this well and include the rails link you provided in our documentation.

I am debating about rescuing `ArgumentError` here in cases where the merchant passes an invalid value, but the downside of this would be that we end up silently substituting a default value that is probably not the merchant's actual time zone. So I suppose we can just leave it how you have it and hopefully document it well enough that it doesn't cause any issues.

Reply:
Sounds good! I will update the documentation as well 👍